pyseir/cli.py

```diff
@@ -137,51 +137,138 @@ def _map_outputs(
+    # )
+
+
+def _build_all_for_states(
+    states=[],
+    run_mode=DEFAULT_RUN_MODE,
+    generate_reports=True,
+    output_interval_days=4,
     skip_download=False,
-    states_only=False,
     output_dir=None,
     skip_whitelist=False,
 ):
-
+    # prepare data
     _cache_global_datasets()
-
     if not skip_download:
         cache_all_data()
-
     if not skip_whitelist:
         _generate_whitelist()
 
+    # run states in paralell
+    all_county_fips = []
+    for state in states:
+        all_county_fips += model_fitter.build_county_list(state)
+
+    # do everything for just states in paralell
+    states_only_func = partial(
+        _state_only_pipeline,
+        run_mode=run_mode,
+        generate_reports=generate_reports,
+        output_interval_days=output_interval_days,
+        states_only=True,
+        output_dir=output_dir,
+    )
+    p = Pool()
+    p.map(states_only_func, states)
+    p.close()
+    p.join()
+
+    #calculate culate inference
+    # parallelize async
+    p = Pool()
+    for state in states:
+        p.apply_async(_infer_rt, state)
+    p.close()
+    p.join()
+
+    #calculate ensemble
+    print(f"executing model for {len(all_county_fips)} counties")
+    p = Pool()
+    p.map(model_fitter._execute_model_for_fips, all_county_fips)
+    p.close()
+    p.join()
+
+    #calculate ensemble
+    print(f"running model for {len(all_county_fips)} counties")
+    p = Pool()
+    ensemble_func = partial(_run_county, ensemble_kwargs=dict(run_mode=run_mode, generate_report=generate_reports))
+    p.map(ensemble_func, all_county_fips)
+    p.close()
+    p.join()
+
+    #output it all
+    output_interval_days = int(output_interval_days)
+    _cache_global_datasets()
+
+    print(f"outputing web results for states and {len(all_county_fips)} counties")
+    #p = Pool()
+    for state in states:
+        web_ui_mapper = WebUIDataAdaptorV1(
+            state,
+            output_interval_days=output_interval_days,
+            run_mode=run_mode,
+            jhu_dataset=nyt_dataset,
+            cds_dataset=cds_dataset,
+            output_dir=output_dir,
+        )
+        web_ui_mapper.generate_state(all_fips=all_county_fips)
+        # mapper_list += build_function_with_fips
+        #web_ui_mapper.build_own_fips(all_county_fips)
+        #web_ui_mapper.execute_own_fips_async(p)
+
+    # p.close()
+    # p.join()
+
+    return
+
+
+def _run_all(
+    state=None,
+    run_mode=DEFAULT_RUN_MODE,
+    generate_reports=False,
+    output_interval_days=1,
+    skip_download=False,
+    states_only=False,
+    output_dir=None,
+    skip_whitelist=False,
+):
     if state:
         # Deprecate temporarily since not needed. Our full model fits have
         # superseded these for now. But we may return to a context where this
         # method is used to measure localized Reff.
         # if not states_only:
         #     _impute_start_dates(state)
-        _infer_rt(state, states_only=states_only)
```

Comment:
Might suggest we keep both run method until we have verified everything and added new docs.